python/cudf/cudf/tests/test_datetime.py

```diff
@@ -397,3 +397,39 @@ def test_datetime_nunique(data, nulls):
     expected = psr.nunique()
     got = gsr.nunique()
     assert_eq(got, expected)
+
+
+testdata = [
+    (
+        Series(
+            ["2018-01-01", None, "2019-01-31", None, "2018-01-01"],
+            dtype="datetime64[ms]",
+        ),
+        True,
+    ),
+    (
+        Series(
+            [
+                "2018-01-01",
+                "2018-01-02",
+                "2019-01-31",
+                "2018-03-01",
+                "2018-01-01",
+            ],
+            dtype="datetime64[ms]",
+        ),
+        False,
+    ),
+]
```

Comment:
Can you add a condition of a pyarrow array specifically constructed with a `NaT` and a `null`?
```
pa.array([0, np.iinfo('int64').min, np.iinfo('int64').max, None], type=pa.timestamp('ns'))
```

Reply:
This one is failing as everything is being taken care from `PyArrow` and PyArrow doesn't consider `np.iinfo('int64').min` as `NaT`. So, it fails.

Reply:
I wouldn't expect `PyArrow` to consider it to be `NaT` and I wouldn't expect us to consider it to be `NaT` or `null` either.

Reply:
ok